api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -405,6 +365,43 @@ private DataSetRequest generateDataSetRequestFromPreviewRequest(
     return ResponseEntity.ok(previewQueryResponse);
   }
 
+  private void addFieldValuesFromBigQueryToPreviewList(List<DataSetPreviewValueList> valuePreviewList, FieldValueList fieldValueList) {
+    IntStream.range(0, fieldValueList.size())
+        .forEach(
+            columnNumber -> {
+              valuePreviewList
+                  .get(columnNumber)
+                  .addQueryValueItem(
+                      Optional.ofNullable(fieldValueList.get(columnNumber).getValue())
+                          .orElse("")
+                          .toString());
+            });
+  }
+
+  private void formatTimestampValues(List<DataSetPreviewValueList> valuePreviewList, Field fields) {
+    DataSetPreviewValueList previewValue =
+        valuePreviewList.stream()
+            .filter(preview -> preview.getValue().equalsIgnoreCase(fields.getName()))
+            .findFirst()
+            .orElseThrow(() -> new ServerErrorException("Value should be present when it is not in dataset preview request"));
+    if (fields.getType() == LegacySQLTypeName.TIMESTAMP) {
+      List<String> queryValues = new ArrayList<>();
+      DateFormat dateFormat = new SimpleDateFormat(dateFormatString);
+      previewValue
+          .getQueryValue()
+          .forEach(
+              value -> {
+                if (!value.equals(emptyCellMarker)) {
+                  Double fieldValue = Double.parseDouble(value);
+                  queryValues.add(dateFormat.format(new Date(fieldValue.longValue())));
```

Comment:
nit: I think this works, but it seems like we'd want all the controllers to use the same data formatting function so the UI knows what to expect.

Reply:
This is currently the only place we are using date formatting. I'd push back on abstracting until we have a second date formatting location.

Reply:
That's interesting. Is this really the only place in all our UI we need to get date strings from the backend? That's a different smell, but for another day I suppose. Next time we're looking at the request structure, maybe we can just pass epoch millis as a long and let the FE deal with it.

Reply:
I think maybe swagger does some work on this, but when I searched for `.format` this was all I found (other than string formatting)